src/mlpack/bindings/python/print_wrapper_py.cpp

```diff
@@ -0,0 +1,435 @@
+ */
+#include "print_wrapper_py.hpp"
+#include "get_arma_type.hpp"
+#include "wrapper_functions.hpp"
+#include <mlpack/core/util/io.hpp>
+
+using namespace mlpack::util;
+using namespace std;
+
+namespace mlpack {
+namespace bindings {
+namespace python {
+
+void PrintWrapperPY(const std::string& category,
+                    const std::string& groupName,
+                    const std::string& validMethods)
+{
+  map<string, Params> params;
+  set<string> serializable; // all serializable parameters.
+  map<string, bool> hyperParams; // if a parameter is hyperparameter.
+  map<string, bool> isBool; // if a parameter is a bool.
+  vector<string> methods = GetMethods(validMethods);
+  int indent = 0; // keeps track of indentation at each point.
+
+  for(auto i: methods)
+  {
+    params[i] = IO::Parameters(groupName + "_" + i);
+  }
+
+  if(category == "regression")
+  {
+    cout << "class BaseEstimator:" << endl;
+    cout << "  pass" << endl << endl;
+  }
+  else if(category == "classification")
+  {
+    cout << "class BaseEstimator:" << endl;
+    cout << "  pass" << endl << endl;
+    cout << "class ClassifierMixin:" << endl;
+    cout << "  pass" << endl << endl;
+  }
+
+  // Import different mlpack programs that are to be wrapped.
+  for(int i=0; i<methods.size(); i++)
+  {
+    cout << "from mlpack." << groupName << "_" << methods[i] << " ";
+    cout << "import " << groupName << "_" << methods[i] << endl;
+  }
+
+  // Try importing scikit-learn, only for classification and
+  // regression.
+  if(category == "regression" || category == "classification")
+  {
+    cout << "import warnings" << endl;
+    cout << "warnings.filterwarnings(\"default\"";
+    cout << ", category=ImportWarning)" << endl;
+  }
+
+  if(category == "regression")
+  {
+    cout << "try:" << endl;
+    cout << "  from sklearn.base import BaseEstimator" << endl;
+  }
+  else if(category == "classification")
+  {
+    cout << "try: " << endl;
+    cout << "  from sklearn.base import BaseEstimator, ClassifierMixin" << endl;
+  }
+
+  if(category == "regression" || category == "classification")
+  {
+    cout << "except:" << endl;
+    cout << "  warnings.warn(\"scikit learn not present:\\n";
+    cout << "  this mlpack method supports scikit utils, install scikit-learn\\n";
+    cout << "  if you wish to use them!\", category=ImportWarning)" << endl;
+    cout << endl;
+  }
+
+  // Check and store if every parameter is serializable,
+  // a hyperparameter and boolean.
+  for(auto i=params.begin(); i!=params.end(); i++)
+  {
+    map<string, ParamData> methodParams = i->second.Parameters();
+
+    for(auto itr=methodParams.begin(); itr!=methodParams.end(); itr++)
+    {
+      // Checking for serializability.
+      bool isSerial;
+      i->second.functionMap[itr->second.tname]["IsSerializable"](
+          itr->second, NULL, (void*)& isSerial);
+      if(isSerial)
+      	serializable.insert(itr->second.cppType);
```

Comment:
```suggestion
      if (isSerial)
        serializable.insert(itr->second.cppType);
```
Another little style issue I found. :)